mex.go

```diff
@@ -446,6 +464,31 @@ func (mexset *messageExchangeSet) forwardPeerFrame(frame *Frame) error {
 	return nil
 }
 
+func (mexset *messageExchangeSet) handleCancel(frame *Frame) {
+	if mexset.shutdown {
```

Comment:
a lock is acquired when it's set, so a read lock should be acquired here too?

Reply:
Shouldn't need this actually, this method should look a lot like `forwardPeerFrame` above, which doesn't have this, so reverted it.